### ISSUE TITLE
How to provide concrete names for generic `async_graphql::Object`?

### ISSUE THREAD

**Issue by elertan** [async-graphql/async-graphql#633]
The documentation shows to to provide concrete names for `async_graphql::SimpleObject`, but this does not seem to work for `async_graphql::Object`.

From docs for simple object:
```rs
#[derive(SimpleObject)]
#[graphql(concrete(name = "SomeName", params(SomeType)))]
#[graphql(concrete(name = "SomeOtherName", params(SomeOtherType)))]
pub struct SomeGenericObject<T: OutputType> {
    field1: Option<T>,
    field2: String
}
```

Attempts:
```rs
#[graphql(concrete(name = "SomeName", params(SomeType)))]
#[graphql(concrete(name = "SomeOtherName", params(SomeOtherType)))]
pub struct SomeGenericObject<T: OutputType> {
    field1: Option<T>,
    field2: String
}

#[Object]
impl<T> SomeGenericObject<T> where T: OutputType {
    async fn field1(&self) -> Option<&T> { ... }
}
```

or

```rs
pub struct SomeGenericObject<T: OutputType> {
    field1: Option<T>,
    field2: String
}

#[Object]
#[graphql(concrete(name = "SomeName", params(SomeType)))]
#[graphql(concrete(name = "SomeOtherName", params(SomeOtherType)))]
impl<T> SomeGenericObject<T> where T: OutputType {
    async fn field1(&self) -> Option<&T> { ... }
}
```

or

```rs
pub struct SomeGenericObject<T: OutputType> {
    field1: Option<T>,
    field2: String
}

#[graphql(concrete(name = "SomeName", params(SomeType)))]
#[graphql(concrete(name = "SomeOtherName", params(SomeOtherType)))]
#[Object]
impl<T> SomeGenericObject<T> where T: OutputType {
    async fn field1(&self) -> Option<&T> { ... }
}
```
All do not compile.

How would one provide these names?
Thanks!

**Comment by sunli829** [comment]
The `Object` macro does not support the `concrete` attribute because it is not needed.
You can do it like this:

```rust
    struct MyObj<T> {
         value: T,
     }

     #[Object(name = "MyObjI32")]
     impl MyObj<i32> {
         async fn value(&self) -> i32 {
             self.value
         }
     }

     #[Object(name = "MyObjBool")]
     impl MyObj<bool> {
         async fn value(&self) -> bool {
             self.value
         }
     }
```

**Comment by elertan** [comment]
Hi @sunli829,

Thanks for the quick reply.

That makes sense, seems like a good solution for implementations where the logic would differ depending on the concrete type that would be passed in.
But sadly this still does not solve the issue I am having, because in my case the implementation is exactly the same for each of the concrete types I would pass in:

```rs
use async_graphql::*;

struct MyObj<T> where T: OutputType {
    value: T,
}

// I don't want to duplicate this for each concrete type...
// My real code is more complex than this simple example
#[Object]
impl<T> MyObj<T> where T: OutputType {
    async fn value(&self) -> &T {
        self.value
    }
}

#[SimpleObject]
struct TypeA {
    field_a: u32,
}

#[SimpleObject]
struct TypeB {
    field_b: String,
}

#[graphql(name = "TypeAMyObj")]
type TypeAMyObj = MyObj<TypeA>;

#[graphql(name = "TypeBMyObj")]
type TypeBMyObj = MyObj<TypeB>;

struct QueryRoot;

#[Object]
impl QueryRoot {
    async fn a(&self) -> TypeAMyObj {
        todo!()
    }
    async fn b(&self) -> TypeBMyObj {
        todo!()
    }
}
```

I wouldn't want to duplicate the impl block as stated in the comment...
If I had to go down that route then using macros would be a solution, but it seems like nothing generics wouldn't be able to solve?

Again thanks for your help!

**Comment by sunli829** [comment]
Maybe it is not easy to support this, but I will try it. 😁

**Comment by elertan** [comment]
Awesome!
For now I have used a macro to generate the object implementations with the name attribute provided, works fine for now 😄 .

**Comment by sunli829** [comment]
@elertan I added the `concrete` attribute to the `Object` macro in `master` branch, please try it. 🙂

here is the test code:

https://github.com/async-graphql/async-graphql/blob/9e69b40987d9b9e044ccc329c88912f945983ac6/tests/generic_types.rs#L296

**Comment by sunli829** [comment]
Released in `2.10.2`
